perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2099,6 +2085,120 @@ public void sendPasswordResetLinkEmail(PerunSession sess, Member member, String
+		String loginAttributeName = PasswordManagerModule.LOGIN_PREFIX + namespace;
+		//create new user
+		User sponsoredUser = getPerunBl().getUsersManagerBl().createUser(session,parseUserFromCommonName(guestName));
+
+		//create the user account in external system
+		Map<String, String> p = new HashMap<>();
+		p.put(PasswordManagerModule.TITLE_BEFORE_KEY,sponsoredUser.getTitleBefore());
+		p.put(PasswordManagerModule.FIRST_NAME_KEY,sponsoredUser.getFirstName());
+		p.put(PasswordManagerModule.LAST_NAME_KEY,sponsoredUser.getLastName());
+		p.put(PasswordManagerModule.TITLE_AFTER_KEY,sponsoredUser.getTitleAfter());
+		p.put(PasswordManagerModule.PASSWORD_KEY,password);
+		Map<String, String> r = getPerunBl().getUsersManagerBl().generateAccount(session, namespace, p);
+		String login = r.get(loginAttributeName);
+		setLoginToSponsoredUser(session,sponsoredUser,loginAttributeName,login);
+
+		//create the member in Perun
+		Member sponsoredMember = getMembersManagerImpl().createSponsoredMember(session, vo, sponsoredUser, sponsor);
+		getPerunBl().getAuditer().log(session, "{} created, sponsored by {}", sponsoredMember, sponsor);
+		extendMembership(session, sponsoredMember);
+		insertToMemberGroup(session, sponsoredMember, vo);
+		validateMemberAsync(session, sponsoredMember);
+		getPerunBl().getUsersManagerBl().validatePasswordAndSetExtSources(session, sponsoredUser, login, namespace);
+		return sponsoredMember;
+	}
+
+	private void setLoginToSponsoredUser(PerunSession sess, User sponsoredUser, String loginAttributeName, String login) throws InternalErrorException {
+		try {
+			Attribute a = getPerunBl().getAttributesManagerBl().getAttribute(sess, sponsoredUser, loginAttributeName);
+			a.setValue(login);
+			getPerunBl().getAttributesManagerBl().setAttribute(sess,sponsoredUser,a);
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("attribute {} not defined",loginAttributeName);
+		} catch (WrongAttributeValueException | WrongReferenceAttributeValueException e) {
+			log.error("cannot set attribute {} for sponsored user {}",loginAttributeName,sponsoredUser.getId());
+		}
+	}
+
+	@Override
+	public Member sponsorMember(PerunSession session, Vo vo, User sponsored, User sponsor) throws MemberNotExistsException, InternalErrorException, AlreadyMemberException {
+		Member sponsoredMember;
+		try {
+			sponsoredMember = getMemberByUser(session, vo, sponsored);
+			if(!sponsoredMember.isSponsored()) {
+				throw new IllegalArgumentException("member "+sponsoredMember.getId()+" is not marked as sponsored");
```

Comment:
it is a checked exception. It is cz.metacentrum.perun.core.api.exceptions.IllegalArgumentException